src/priceClient/priceClient.ts

```diff
@@ -0,0 +1,212 @@
+import assert from "assert";
+import { CoinGeckoPrice } from "../coingecko";
+import { Logger } from "../relayFeeCalculator"; // @todo: Relocate Logger to utils?
+
+export { Logger } from "../relayFeeCalculator"; // Permit adapters to import a common definition.
+export type TokenPrice = CoinGeckoPrice; // Temporary inversion; CoinGecko should source from here.
+
+// Represents valid source for spot prices (Across API, CoinGecko, ...)
+export interface PriceFeedAdapter {
+  readonly name: string;
+  getTokenPrice(address: string, currency: string, platform: string): Promise<TokenPrice>;
+  getTokenPrices(addresses: string[], currency: string, platform: string): Promise<TokenPrice[]>;
+}
+
+export type PriceCache = {
```

Comment:
The name PriceCache is a bit unfortunate because it's used to describe a dictionary containing TokenPrice types and has broader usage than in caching. Would like to find a better name for it.